optuna/storages/rdb/storage.py

```diff
@@ -326,14 +326,53 @@ def get_all_study_summaries(self):
 
         return study_sumarries
 
-    def create_new_trial_id(self, study_id):
-        # type: (int) -> int
+    def create_new_trial_id(self, study_id, template_trial=None):
+        # type: (int, Optional[structs.FronzenTrial]) -> int
 
         session = self.scoped_session()
 
-        trial = models.TrialModel(study_id=study_id, state=structs.TrialState.RUNNING)
+        if template_trial is None:
+            trial = models.TrialModel(study_id=study_id, state=structs.TrialState.RUNNING)
+        else:
+            # Because only `RUNNING` trials can be updated,
+            # we temporarily set the state of the new trial to `RUNNING`.
+            # After all fields of the trial have been updated,
+            # the state is set to `template_trial.state`.
+            temp_state = structs.TrialState.RUNNING
+
+            trial = models.TrialModel(
+                study_id=study_id,
+                state=temp_state,
+                value=template_trial.value,
+                datetime_start=template_trial.datetime_start,
+                datetime_complete=template_trial.datetime_complete,
+            )
 
         session.add(trial)
+        session.flush()
+
+        if template_trial is not None:
+            for param_name, param_value in template_trial.params.items():
+                distribution = template_trial.distributions[param_name]
+                param_value_in_internal_repr = distribution.to_internal_repr(param_value)
+                self._set_trial_param_without_commit(session, trial.trial_id, param_name,
+                                                     param_value_in_internal_repr, distribution)
+
+            for key, value in template_trial.user_attrs.items():
+                self._set_trial_user_attr_without_commit(session, trial.trial_id, key, value)
+
+            for key, value in template_trial.system_attrs.items():
+                if key == '_number':
+                    continue
```

Comment:
The test cases for `create_new_trial_id` does not cover this condition. Please input `template_trial` which has `_number` key in `system_attrs` in the test case.

Reply:
I updated ` test_create_new_trial_id_with_template_trial` test case for covering this line (https://github.com/pfnet/optuna/pull/464/commits/25687d2ee81d1abef5a3567c779a73cdfe12159b).